docs(http): add examples for post request

jsonRUHLS opened the pull request:
add examples to the docs examples for http post requests. We already have docs examples for http get requests

github issue https://github.com/angular/angular/issues/25168

## PR Checklist
Please check if your PR fulfills the following requirements:

- [X] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [X] Tests for the changes have been added (for bug fixes / features)
- [X] Docs have been added / updated (for bug fixes / features)


## PR Type
What kind of change does this PR introduce?

<!-- Please check the one that applies to this PR using "x". -->

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [X] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:


## What is the current behavior?
<!-- Please describe the current behavior that you are modifying, or link to a relevant issue. -->

Issue Number: https://github.com/angular/angular/issues/25168


## What is the new behavior?

Add docs examples for HTTP post requests

## Does this PR introduce a breaking change?

- [ ] Yes
- [X] No


<!-- If this PR contains a breaking change, please describe the impact and migration path for existing applications below. -->


## Other information

jsonRUHLS committed:
docs(http): add examples for post request

add examples to the docs examples for http post requests. We already have docs examples for http get requests

github issue https://github.com/angular/angular/issues/25168

jsonRUHLS committed:
docs(http): remove whitespace

remove whitespace that is breaking lint

github issue: https://github.com/angular/angular/issues/25168

trotyl commented on aio/content/guide/http.md:
````diff
@@ -439,13 +440,14 @@ In fact, each `subscribe()` initiates a separate, independent execution of the o
 
 ```javascript
-const req = http.get<Heroes>('/api/heroes');
+const req = http.get < Heroes > "/api/heroes";
````
What is this change?

trotyl commented on aio/content/guide/http.md:
```diff
@@ -605,9 +607,10 @@ If an interceptor could modify the original request object, the re-tried operati
-  // Typescript disallows the following assignment because req.url is readonly
-  req.url = req.url.replace('http://', 'https://');
+// Typescript disallows the following assignment because req.url is readonly
+req.url = req.url.replace("http://", "https://");
```
Single quote is used consistently in all examples, should not be changed.

jsonRUHLS committed:
docs(http): address requests

conform to normal single quote pattern for strings, remove weird change

github issue: https://github.com/angular/angular/issues/25168

jsonRUHLS committed:
docs(http): address requests

conform to normal single quote pattern for string, remove weird change

github issue: https://github.com/angular/angular/issues/25168

jsonRUHLS committed:
docs(http): update dependency

updated typescript to use 3.6.4

github issue: https://github.com/angular/angular/issues/25168

jsonRUHLS committed:
docs(http): update typescript

update typescript dependency to ^3.6.4

github issue: https://github.com/angular/angular/issues/25168

jsonRUHLS committed:
Merge branch 'master' into jsonruhls-docs-http-post-request-mocks-backend

gkalpak commented on aio/content/guide/http.md:
```diff
@@ -160,11 +160,11 @@ Now `HttpClient.get()` returns an `Observable` of type `HttpResponse` rather tha
+region="showConfigResponse"
+header="app/config/config.component.ts (showConfigResponse)"
+
+> </code-example>
```
This formatting is kind if weird (below too). Can you change it to something like:

```html
<code-example
    path="http/src/app/config/config.component.ts"
    region="showConfigResponse"
    header="app/config/config.component.ts (showConfigResponse)">
</code-example>
```

gkalpak commented on aio/content/guide/http.md:
```diff
@@ -1044,6 +1048,14 @@ Now you can write a test that expects a GET Request to occur and provides a mock
   header="app/testing/http-client.spec.ts(httpClient.get)">
 </code-example>
 
+Now you can write a test that expects a POST Request to occur and provides a mock response.
```
"Now" sounds a bit out-of-place here. Maybe something like:

> You can also write a test...

gkalpak commented on aio/content/examples/http/src/testing/http-client.spec.ts:
```diff
@@ -29,33 +30,34 @@ describe('HttpClient testing', () => {
+    // The following `expectOne()` will match the request's URL.
+    // If no requests or multiple requests matched that URL
+    // `expectOne()` would throw.
+    const req = httpTestingController.expectOne('/data');
```
- Please, use `testUrl` here too (for consistency).
- In order for `req` to be available outside of this function (e.g. inside the test-cases), it needs to be declared in the enclosing function scope (e.g. besides `httpTestingController`). But, tbh, I don't feel `beforeEach()` is a good place for such a call. I would move it inside the respective test-cases.

gkalpak commented on aio/content/examples/http/src/testing/http-client.spec.ts:
```diff
@@ -14,6 +14,7 @@ interface Data {
   name: string;
 }
 
+const testData: Data = {name: 'Test Data'};
```
There is little point in "globally" storing murable data like this. Please, move inside the test-cases that use this.

gkalpak commented on aio/content/examples/http/src/testing/http-client.spec.ts:
```diff
@@ -67,8 +69,30 @@ describe('HttpClient testing', () => {
+    req.flush(testData);
+
+    // Finally, assert that there are no outstanding requests.
+    httpTestingController.verify();
```
Not directly related to this PR, but these calls seem to be superfluous inside the `it()` blocks, because there is an `afterEach()` call that takes care `httpTestingController.verify()` for each test.

gkalpak commented on package.json:
```diff
@@ -116,7 +116,7 @@
     "tslib": "^1.10.0",
     "tslint": "5.7.0",
-    "typescript": "~3.6.4",
+    "typescript": "^3.6.4",
```
This (and the associated change in `yarn.lock`) seems unnecessary. Please, remove.

gkalpak reviewed (changes requested):
Thx for working on this, @jsonRUHLS ✨
I left a few comments. Can you also squash all commits into one and use a commit message that follows [our guidelines][1]. Something like:

> docs(http): add example for testing post requests
>
> Fixes #25168

[1]: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#-commit-message-guidelines

AndrewKushnir commented:
FYI, closing this PR due to the lack of response and a number of merge conflicts. Please reopen when you get a chance to look at the changes again. Thank you.
